compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/InteractionUIView.uikit.kt

```diff
@@ -396,30 +434,46 @@ private class GestureRecognizerHandlerImpl(
     private fun checkPanIntent() {
         if (isLocationDeltaAboveSlop) {
             gestureRecognizer?.cancelFailure()
+
+            // When this gesture state transits to UIGestureRecognizerStateBegan, the gesture
+            // iOS stops withholding the intercepted touches and prevents them from being sent
+            // to the interop view.
             gestureRecognizerState = UIGestureRecognizerStateBegan
         }
     }
 
     /**
-     * Stops tracking the given touches. If there are no tracked touches left, reset the initial
-     * location to null.
+     * Stops tracking the given touches associated with [UIEvent]. If those are the last touches,
+     * end the gesture and reset internal state.
      */
     private fun stopTrackingTouches(touches: Set<*>) {
         for (touch in touches) {
```

Comment:
This function still looks affected, but without usages yet. Could you please add an explicit check here?
```suggestion
        if (trackedTouches === touches) {
            trackedTouches.clear()
            return
        }
        for (touch in touches) {
```

Also it iooks like only this who lines actually needed. But I'm OK with cleaning up the stuff a bit

Reply:
`trackedTouches` is never passed now to `stopTrackingTouches` (the actual exception reason).

The commit you proposed is not enough, because we still need to reset internal state.
The idea is that `stopTrackingTouches` is called due to `UIEvent` and `stopTrackingAllTouches` is called by us due to scheduled failure. I wanted to reflect those two separate flows explicitly.